### PR TITLE
add configuration for travisCI and coveralls

### PR DESCRIPTION
**Pull request by zimmski**
This adds support for TravisCI [https://travis-ci.org/] and Coveralls [coveralls.io] to the project which is quite common on Github. Every commit and PR will run through the CI and will be therefore tested automatically. The status of every run will be shown in the Github UI. Coverage reports for every project commit (NOT for PR commits) will be send to Coveralls, so keeping track of the project's coverage is possible.

The code coverage part in the script uses ginkgo and gover to allow recursive coverage report. This is a workaround until -coverprofile can be done on multiple packages. I know that the project has currently only one package but this is future proof and does not hurt since it is automated.

Since I am not the maintainer of the project @dsymonds you have to enable support for the project on TravisCI and Coveralls and add the secure token for Coveralls (you can also post it here and I include it in the commit) http://docs.travis-ci.com/user/encryption-keys/

### CODE REVIEW

**Comment by zimmski** [comment]
@dsymonds  I am looking through my open PRs. Is there any reason this cannot be merged? I can see that https://github.com/golang/oauth2 is using Travis too so it is clearly no "the golang group does not do that"-thing.

Having a CI to check that every commit passes all tests and works with the latest dependencies is a good thing in my opinion. Making the coverage available could also attract newbies to contribute to the project.

**Comment by dsymonds** [comment]
I don't have anything against Travis, just unfamiliarity. It didn't seem particularly important, so I put it on my backlog. I'll take another look.

**Comment by dsymonds** [comment]
Let's strip this back to just continuous integration. Drop the gofmt and coverage stuff, since that's just adding extra complexity here at this stage. Can you do something as simple as https://github.com/golang/appengine/blob/master/.travis.yml?

**Comment by zimmski** [comment]
I reduced it to just testing the project and squashed all commits. Can we add gofmt/coverage later?

The appengine Travis configuration is a little complicated (should I send a PR to them?):
- Defining the Go version should not be needed. I am currently in a discussion with the Travis support. If there is no explicit Go version defined, it should always select the latest stable Go release. (As of today it is 1.4.1 btw before it was the latest 1.3)
- I don't know what they are doing with GOPATH. Since GOPATH is defined if you select Go as the language. And the current folder is always the root of the repository. So a "go get -t -v ./..." should be sufficient.
- Shouldn't a test with the race flag on include every test when the flag is off?

**Comment by dsymonds** [comment]
We can think about coverage later. I don't think this code gets so much churn that gofmt-checking is such a big deal.

I think the appengine config was slightly complex because we wanted to pin it to the version actually being used on App Engine, which isn't always the latest stable.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+language: go
+
+install:
+  - go get -t -v ./...
+
+script:
+  - go test -race -v ./...
```
Comment (dsymonds) [inline .travis.yml]: Actually, for this repo there's so little concurrency that I don't think `-race` is needed. Let's just drop it.

Comment (zimmski) [inline .travis.yml]: ok, done

**Comment by dsymonds** [comment]
LGTM

**Pull request closed without merging**